### PR TITLE
doc: Updated index.md and readme.md

### PR DESCRIPTION
**Pull request by FrancescoSer**
Updated index.md and readme.md:
- Removed outdated info
- Added info about newly added ports
- Fixed a few style issues.
- Rephrased a few sections.

Signed-off-by: Francesco Servidio <francesco.servidio@nordicsemi.no>

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -26,52 +34,48 @@ for more details.
...
 
-The issues being planned and worked on are tracked using GitHub issues. To participate
-please visit:
-
-[MCUboot Issues](https://github.com/mcu-tools/mcuboot/issues)
-
-~~Issues were previously tracked on [MCUboot JIRA](https://runtimeco.atlassian.net/projects/MCUB/summary)
-, but it is now deprecated.~~
+The issues being planned and worked on are tracked using GitHub issues. To
+give your input, visit [MCUBoot GitHub
```
Comment (gustavonihei) [inline docs/index.md:58]:
```suggestion
give your input, visit [MCUboot GitHub
```

File: README.md [hunk 2]
```diff
@@ -23,55 +23,57 @@ define a common infrastructure for the bootloader, system flash layout on
...
 ## Roadmap
 
 The issues being planned and worked on are tracked using GitHub issues. To
-participate please visit:
-
-[MCUboot GitHub Issues](https://github.com/mcu-tools/mcuboot/issues)
-
-~~Issues were previously tracked on [MCUboot JIRA](https://runtimeco.atlassian.net/projects/MCUB/summary)
-, but it is now deprecated.~~
+give your input, visit [MCUBoot GitHub
```
Comment (gustavonihei) [inline README.md:54]:
```suggestion
give your input, visit [MCUboot GitHub
```

File: README.md [hunk 3]
```diff
@@ -23,55 +23,57 @@ define a common infrastructure for the bootloader, system flash layout on
...
+You can find additional documentation on the bootloader in the source files.
+For more information, use the following links:
 
 - [boot/bootutil](boot/bootutil): The core of the bootloader itself.
 - [boot/boot\_serial](boot/boot_serial): Support for serial upgrade within the bootloader itself.
-- [boot/zephyr](boot/zephyr): Port of the bootloader to Zephyr
-- [boot/mynewt](boot/mynewt): Mynewt bootloader app
+- [boot/zephyr](boot/zephyr): Port of the bootloader to Zephyr.
+- [boot/mynewt](boot/mynewt): Bootloader application for Mynewt.
 - [boot/nuttx](boot/nuttx): Bootloader application and port of MCUboot interfaces for NuttX.
```
Comment (gustavonihei) [inline README.md:66]:
```suggestion
- [boot/nuttx](boot/nuttx): Bootloader application and port of MCUboot interfaces for Apache NuttX.
```
Just to keep consistent with the new changes from this PR.

Comment (gustavonihei) [inline README.md:66]:
```suggestion
- [boot/mynewt](boot/mynewt): Bootloader application for Apache Mynewt.
- [boot/nuttx](boot/nuttx): Bootloader application and port of MCUboot interfaces for Apache NuttX.
```
Just to keep consistent with the new changes from this PR.

File: docs/index.md [hunk 4]
```diff
@@ -26,52 +34,48 @@ for more details.
 - [Bootloader design](design.md)
 - [Encrypted images](encrypted_images.md)
 - [imgtool](imgtool.md) - image signing and key management
-- [ecdsa](ecdsa.md) - Information about ECDSA signature formats
+- [ECDSA](ecdsa.md) - nformation about ECDSA signature formats
```
Comment (gustavonihei) [inline docs/index.md:37]:
```suggestion
- [ECDSA](ecdsa.md) - Information about ECDSA signature formats
```

File: docs/index.md [hunk 5]
```diff
@@ -26,52 +34,48 @@ for more details.
...
+You can find additional documentation on the bootloader in the source files.
+For more information, use the following links:
 - [boot/bootutil](https://github.com/mcu-tools/mcuboot/tree/main/boot/bootutil): The core of the bootloader itself.
 - [boot/boot\_serial](https://github.com/mcu-tools/mcuboot/tree/main/boot/boot_serial): Support for serial upgrade within the bootloader itself.
-- [boot/zephyr](https://github.com/mcu-tools/mcuboot/tree/main/boot/zephyr): Port of the bootloader to Zephyr
-- [boot/mynewt](https://github.com/mcu-tools/mcuboot/tree/main/boot/mynewt): Mynewt bootloader app
-- [boot/mbed](https://github.com/mcu-tools/mcuboot/tree/main/boot/mbed): Port of the bootloader to Mbed-OS
+- [boot/zephyr](https://github.com/mcu-tools/mcuboot/tree/main/boot/zephyr): Port of the bootloader to Zephyr.
+- [boot/mynewt](https://github.com/mcu-tools/mcuboot/tree/main/boot/mynewt): Bootloader application for Mynewt.
+- [boot/nuttx](https://github.com/mcu-tools/mcuboot/tree/main/boot/nuttx): Bootloader application and port of MCUboot interfaces for NuttX.
```
Comment (gustavonihei) [inline docs/index.md:69]:
```suggestion
- [boot/mynewt](https://github.com/mcu-tools/mcuboot/tree/main/boot/mynewt): Bootloader application for Apache Mynewt.
- [boot/nuttx](https://github.com/mcu-tools/mcuboot/tree/main/boot/nuttx): Bootloader application and port of MCUboot interfaces for Apache NuttX.
```

File: docs/index.md (hunk 1 above)
Comment (FrancescoSer) [reply docs/index.md]: fixed

File: README.md (hunk 2 above)
Comment (FrancescoSer) [reply README.md]: fixed

File: docs/index.md (hunk 4 above)
Comment (FrancescoSer) [reply docs/index.md]: fixed

File: docs/index.md (hunk 5 above)
Comment (FrancescoSer) [reply docs/index.md]: fixed

**Review by gustavonihei — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -23,55 +23,57 @@ define a common infrastructure for the bootloader, system flash layout on
...
-target with a complete port planned.
+MCUboot is operating-system and hardware independent, and relies on
+hardware porting layers from the operating system it works with. Currently,
+MCUboot works with the following operating systems:
+- [Zephyr](https://www.zephyrproject.org/)
+- [Apache Mynewt](https://mynewt.apache.org/)
+- [Apache NuttX](https://nuttx.apache.org/)
+- [RIOT](https://www.riot-os.org/)
+- [Mbed OS](https://os.mbed.com/)
+- [Espressif IDF](https://idf.espressif.com/)
```
Comment (d3zd3z) [inline README.md:34]: We should probably also include the Cypress/Infineon port. https://www.cypress.com/ is their site, their docs are in boot/cypress/README.md

File: docs/index.md [hunk 7]
```diff
@@ -1,22 +1,30 @@
...
+
+MCUboot is operating-system and hardware independent, and relies on
+hardware porting layers from the operating system it works with. Currently
+MCUboot works with the following operating systems:
+- [Zephyr](https://www.zephyrproject.org/)
+- [Apache Mynewt](https://mynewt.apache.org/)
+- [Apache NuttX](https://nuttx.apache.org/)
+- [RIOT](https://www.riot-os.org/)
+- [Mbed OS](https://os.mbed.com/)
+- [Espressif IDF](readme-espressif.md)
```
Comment (d3zd3z) [inline docs/index.md:19]: Again, have Cypress/Infineon in the list.

File: docs/index.md [hunk 8]
```diff
@@ -1,22 +1,30 @@
...
 
+MCUboot is a secure bootloader for 32-bit MCUs.
+
 ## Overview
 
-MCUboot is a secure bootloader for 32-bit MCUs.   The goal of MCUboot is to
-define a common infrastructure for the bootloader, system flash layout on
-microcontroller systems, and to provide a secure bootloader that enables
-easy software upgrade.
+The goal of MCUboot is to define a common infrastructure for the bootloader,
```
Comment (d3zd3z) [inline docs/index.md:7]: Do we still want to mention 32-bits? We are definitely not going to be supporting 8 or 16 bit MCUs.

File: docs/index.md [hunk 9]
```diff
@@ -26,52 +34,48 @@ for more details.
...
+For more information, use the following links:
 - [boot/bootutil](https://github.com/mcu-tools/mcuboot/tree/main/boot/bootutil): The core of the bootloader itself.
 - [boot/boot\_serial](https://github.com/mcu-tools/mcuboot/tree/main/boot/boot_serial): Support for serial upgrade within the bootloader itself.
-- [boot/zephyr](https://github.com/mcu-tools/mcuboot/tree/main/boot/zephyr): Port of the bootloader to Zephyr
-- [boot/mynewt](https://github.com/mcu-tools/mcuboot/tree/main/boot/mynewt): Mynewt bootloader app
-- [boot/mbed](https://github.com/mcu-tools/mcuboot/tree/main/boot/mbed): Port of the bootloader to Mbed-OS
+- [boot/zephyr](https://github.com/mcu-tools/mcuboot/tree/main/boot/zephyr): Port of the bootloader to Zephyr.
+- [boot/mynewt](https://github.com/mcu-tools/mcuboot/tree/main/boot/mynewt): Bootloader application for Apache Mynewt.
+- [boot/nuttx](https://github.com/mcu-tools/mcuboot/tree/main/boot/nuttx): Bootloader application and port of MCUboot interfaces for Apache NuttX.
+- [boot/mbed](https://github.com/mcu-tools/mcuboot/tree/main/boot/mbed): Port of the bootloader to Mbed OS.
```
Comment (d3zd3z) [inline docs/index.md:70]: Let's add Cypress here as well.

**Review by d3zd3z — CHANGES_REQUESTED** [review]

File: docs/index.md (hunk 9 above)
Comment (gustavonihei) [reply docs/index.md:70]: The Espressif port is also missing here:
```suggestion
- [boot/mbed](https://github.com/mcu-tools/mcuboot/tree/main/boot/mbed): Port of the bootloader to Mbed OS.
- [boot/espressif](https://github.com/mcu-tools/mcuboot/tree/main/boot/espressif): Bootloader application and MCUboot port for Espressif SoCs.
```

Comment (FrancescoSer) [reply docs/index.md:72]: fixed

File: docs/index.md (hunk 7 above)
Comment (FrancescoSer) [reply docs/index.md]: fixed

File: docs/index.md (hunk 8 above)
Comment (FrancescoSer) [reply docs/index.md:7]: removed the mention

File: README.md (hunk 6 above)
Comment (FrancescoSer) [reply README.md:34]: fixed

File: docs/index.md [hunk 10]
```diff
@@ -1,77 +1,89 @@
...
+You can find additional documentation on the bootloader in the source files.
+For more information, use the following links:
+- [boot/bootutil](https://github.com/mcu-tools/mcuboot/tree/main/boot/bootutil) - The core of the bootloader itself.
+- [boot/boot\_serial](https://github.com/mcu-tools/mcuboot/tree/main/boot/boot_serial) - Support for serial upgrade within the bootloader itself.
+- [boot/zephyr](https://github.com/mcu-tools/mcuboot/tree/main/boot/zephyr) - Port of the bootloader to Zephyr.
+- [boot/mynewt](https://github.com/mcu-tools/mcuboot/tree/main/boot/mynewt) - Bootloader application for Apache Mynewt.
+- [boot/nuttx](https://github.com/mcu-tools/mcuboot/tree/main/boot/nuttx) - Bootloader application and port of MCUboot interfaces for Apache NuttX.
+- [boot/mbed](https://github.com/mcu-tools/mcuboot/tree/main/boot/mbed) - Port of the bootloader to Mbed OS.
+- [boot/espressif](boot/espressif) - Bootloader application and MCUboot port for Espressif SoCs.
+- [boot/cypress](boot/cypress) - Bootloader application and MCUboot port for Cypress/Infineon SoCs.
```
Comment (gustavonihei) [inline docs/index.md:78]:
```suggestion
- [boot/espressif](https://github.com/mcu-tools/mcuboot/tree/main/boot/espressif) - Bootloader application and MCUboot port for Espressif SoCs.
- [boot/cypress](https://github.com/mcu-tools/mcuboot/tree/main/boot/cypress) - Bootloader application and MCUboot port for Cypress/Infineon SoCs.
```
The link target is incorrect here

File: docs/index.md (hunk 8 above)
Comment (d3zd3z) [reply docs/index.md]: I guess I didn't word that very clearly. I meant to say that we should probably continue to mention 32-bits, as we definitely don't support smaller microcontrollers, and we haven't really tried porting it to 64 bits.

File: docs/index.md (hunk 10 above)
Comment (FrancescoSer) [reply docs/index.md]: thanks, fixed both here and in readme.md

File: docs/index.md (hunk 8 above)
Comment (FrancescoSer) [reply docs/index.md]: Ok, I guess I jumped the gun here. Change reverted.

**Review by pkarashchenko — APPROVED** [review]

File: docs/index.md [hunk 11]
```diff
@@ -1,77 +1,89 @@
...
-  on how to contribute to MCUboot
+  - [Mbed OS](readme-mbed.md)
+  - [Espressif IDF](readme-espressif.md)
+  - [Cypress/Infineon](../boot/cypress/readme.md)
+  - [Simulator](../sim/README.rst)
 - Testing
-  - [Zephyr](testplan-zephyr.md) test plan
-  - [mynewt](testplan-mynewt.md) test plan
+  - [Zephyr](testplan-zephyr.md) - Zephyr test plan
+  - [Mynewt](testplan-mynewt.md) - Mynewt test plan
```
Comment (gustavonihei) [inline docs/index.md:52]:
```suggestion
  - [Apache Mynewt](testplan-mynewt.md) - Apache Mynewt test plan
```

Comment (FrancescoSer) [reply docs/index.md]: fixed

**Comment by FrancescoSer** [comment]
@gustavonihei @d3zd3z Feel free to review this again

**Review by gustavonihei — APPROVED** [review]
LGTM!

**Review by utzig — APPROVED** [review]
LGTM, just needs fixing of signed commit to be possible to merge.

**Comment by gustavonihei** [comment]
@FrancescoSer The author name still differs from the name used in the `signed-off-by` line.

**Review by d3zd3z — APPROVED** [review]

**Pull request merged**